front/pages/api/w/[wId]/vaults/[vId]/data_sources/index.ts

```diff
@@ -0,0 +1,542 @@
+  isDataSourceNameValid,
+  sendUserOperationMessage,
+  WebCrawlerConfigurationTypeSchema,
+} from "@dust-tt/types";
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import config from "@app/lib/api/config";
+import { getDataSource, getDataSources } from "@app/lib/api/data_sources";
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { getOrCreateSystemApiKey } from "@app/lib/auth";
+import {
+  getDefaultDataSourceDescription,
+  getDefaultDataSourceName,
+  isConnectorProviderAllowedForPlan,
+  isConnectorProviderAssistantDefaultSelected,
+  isValidConnectorSuffix,
+} from "@app/lib/connector_providers";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import { DataSourceViewResource } from "@app/lib/resources/data_source_view_resource";
+import { VaultResource } from "@app/lib/resources/vault_resource";
+import { ServerSideTracking } from "@app/lib/tracking/server";
+import { isDisposableEmailDomain } from "@app/lib/utils/disposable_email_domains";
+import logger from "@app/logger/logger";
+import { apiError } from "@app/logger/withlogging";
+
+// Sorcery: Create a union type with at least two elements to satisfy t.union
+const ConnectorProviderCodec = t.union([
+  t.literal(CONNECTOR_PROVIDERS[0]),
+  t.literal(CONNECTOR_PROVIDERS[1]),
+  ...CONNECTOR_PROVIDERS.slice(2).map((provider) => t.literal(provider)),
+]);
+
+const PostManagedDataSourceRequestBodySchema = t.type({
+  provider: ConnectorProviderCodec,
+  connectionId: t.union([t.string, t.undefined]),
+  configuration: ConnectorConfigurationTypeSchema,
+});
+
+const PostStaticDataSourceRequestBodySchema = t.type({
+  name: t.string,
+  description: t.string,
+  assistantDefaultSelected: t.boolean,
+});
+
+type PostVaultDataSourceResponseBody = {
+  dataSourceView: DataSourceViewType;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<PostVaultDataSourceResponseBody>>,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.workspace();
+  const plan = auth.plan();
+  const user = auth.user();
+
+  if (!owner || !plan || !user) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  const vault = await VaultResource.fetchById(auth, req.query.vId as string);
+
+  if (
+    !vault ||
+    (!auth.isAdmin() && !auth.hasPermission([vault.acl()], "write"))
+  ) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "vault_not_found",
+        message: "The vault you requested was not found.",
+      },
+    });
+  }
+
+  const dataSourceEmbedder = owner.defaultEmbeddingProvider ?? "openai";
```

Comment:
No magic variables please, let's have "openai" in a constant which is shared in all places where we need the default embedder.